sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -179,6 +179,7 @@ trait ScalaReflection {
       case t if t <:< definitions.ShortTpe => Schema(ShortType, nullable = false)
       case t if t <:< definitions.ByteTpe => Schema(ByteType, nullable = false)
       case t if t <:< definitions.BooleanTpe => Schema(BooleanType, nullable = false)
+      case other => sys.error(s"Schema for type $other is currently not supported")
```

Comment:
(Yeah I corrected my comment right after I made it -- of course `other` is required here.) I don't feel strongly about it; all else equal `RuntimeException` is not ideal, but it's more important to be consistent with similar code than anything.